src/dany/02_hash/solutions.js

```diff
@@ -0,0 +1,41 @@
+const getBigger2PlayCnt = (playCnts) => {
+  playCnts.sort((a, b) => {
+    return a.cnt < b.cnt ? 1 : a.cnt > b.cnt ? -1 : 0
+  })
+  return [playCnts[0], playCnts[1]]
+}
+
+// 시간복잡도 : O(6n + n log n + n + n) => O(n log n)
+export const bestAlbum = (genres, plays) => {
+  const playPerGenre = {}
+  for (let i = 0; i < genres.length; i++) {
+    const genre = genres[i]
+    const currentPlayCount = playPerGenre[genre]
+    if (!currentPlayCount) {
+      playPerGenre[genre] = {
+        playCnts: [{ id: i, cnt: plays[i] }],
+        totalPlayCnt: plays[i]
+      }
+    } else {
+      playPerGenre[genre].totalPlayCnt = playPerGenre[genre].totalPlayCnt + plays[i]
+      playPerGenre[genre].playCnts = getBigger2PlayCnt([...playPerGenre[genre].playCnts, { id: i, cnt: plays[i] }])
+    }
+  }
+  
+  const results = []
+  for (let genre in playPerGenre) {
+    results.push(playPerGenre[genre])
+  }
+
+  results.sort((a, b) => {
+    return a.totalPlayCnt < b.totalPlayCnt ? 1 : a.totalPlayCnt > b.totalPlayCnt ? -1 : 0
+  })
```

Comment:
```suggestion
 results.sort((a, b) => b.totalPlayCnt - a.totalPlayCnt)
```
내림차순 정렬 시 `return b-a` 로 작성하면 삼항 연산자로 조건을 나누지 않아도 정확히 위의 코드와 동일한 결과를 반환하기 때문에 조건문은 생략하는 것이 실행시간을 조금이나마 줄일 수 있을거 같습니다 :)